core/services/mongodb-sync/src/test/java/com/mongodb/stitch/core/services/mongodb/sync/internal/DataSynchronizerUnitTests.kt

```diff
@@ -0,0 +1,151 @@
+                networkMonitor,
+                authMonitor
+        )
+    }
+
+    @After
+    fun teardown() {
+        val dataSpy = Mockito.spy(dataSynchronizer!!)
+        dataSpy.deleteOneById(namespace, insertedId)
+    }
+
+    @Test
+    fun testInsertOneAndSync() {
+        val dataSpy = Mockito.spy(dataSynchronizer!!)
+
+        insertOneAndSync(dataSpy, BsonDocument())
+
+        assertEquals(dataSpy.getSynchronizedDocuments(namespace).first().documentId, insertedId)
+        assertEquals(dataSpy.getSynchronizedDocuments(namespace).size, 1)
+    }
+
+    @Test
+    fun testFindOneById() {
+        val dataSpy = Mockito.spy(dataSynchronizer!!)
+
+        insertOneAndSync(dataSpy, BsonDocument())
+
+        val doc2 = dataSpy.findOneById(
+                namespace,
+                insertedId,
+                BsonDocument::class.java,
+                CodecRegistries.fromCodecs(BsonDocumentCodec())
+        )
+
+        assertEquals(insertedId, doc2["_id"])
+    }
+
+    @Test
+    fun testDeleteOneById() {
+        val dataSpy = Mockito.spy(dataSynchronizer!!)
+
+        insertOneAndSync(dataSpy, BsonDocument())
+
+        assertEquals(dataSpy.getSynchronizedDocuments(namespace).size, 1)
+
+        dataSpy.deleteOneById(namespace, insertedId)
+
+        assertEquals(dataSpy.getSynchronizedDocuments(namespace).size, 0)
+
+        val namespaceArg = ArgumentCaptor.forClass(MongoNamespace::class.java)
+        val idArg = ArgumentCaptor.forClass(BsonValue::class.java)
+
+        verify(dataSpy).desyncDocumentFromRemote(namespaceArg.capture(), idArg.capture())
+
+        assertEquals(namespaceArg.value, namespace)
+        assertEquals(idArg.value, insertedId)
+    }
+
+    private fun insertOneAndSync(dataSpy: DataSynchronizer,
+                                 doc: BsonDocument,
+                                 conflictResolver: SyncConflictResolver<BsonDocument> = SyncConflictResolver {
+                                     _, _, _ ->
+                                     doc
+                                 },
+                                 changeEventListener: ChangeEventListener<BsonDocument> = ChangeEventListener {
+                                     _, _: ChangeEvent<BsonDocument> ->
+                                 }): BsonValue {
+        dataSpy.insertOneAndSync(
+                namespace,
+                doc,
+                conflictResolver,
+                changeEventListener,
+                BsonDocumentCodec()
+        )
+
+        assertTrue(dataSpy.getSynchronizedDocuments(namespace).size > 0)
+
+        val namespaceArg = ArgumentCaptor.forClass(MongoNamespace::class.java)
+        val idArg = ArgumentCaptor.forClass(BsonValue::class.java)
+        val eventListenerArg = ArgumentCaptor.forClass(changeEventListener::class.java)
+        val documentCodecArg = ArgumentCaptor.forClass(BsonDocumentCodec::class.java)
+
+        verify(dataSpy).watchDocument(
```

Comment:
This seems like an odd thing to watch. This is reaching too far into internal details. I don't think this block of code is really necessary.

Reply:
I do this to capture the id.